### PR TITLE
[ntuple] Make ntpl004_dimuon.C compilable into an executable

### PR DESCRIPTION
**Pull request by eguiraud**
Sometimes it's useful to be able to compile tutorials. Since I had to make these changes to debug something, I'm proposing them here as a PR. Feel free to close it if it's not something you want.

### CODE REVIEW

**Comment by pcanal** [comment]
So this allows to create an executable.  Why was that "much more" convenient than compiling with ACliC (i.e. ```./ntpl004_dimuon``` vs ```root.exe -b -l -q ntpl004_dimuon.C+```?  Both can be used equally well with gdb and valgrind.

**Comment by eguiraud** [comment]
It lets me control compilation options (awkward at best via ACliC) and removes one layer of indirection.

**Comment by pcanal** [comment]
Fair enough.  

Was the `#ifndef __CLING__` really needed?  (i.e. it is not supposed to).

The extra "main" in the tutorials is awkward as it may confused reader on whether it is needed in a macros or not and adding a main in every single tutorials (the logic consequence) would mean technically an ODR violation upon loading (more than one) tutorials .. well actually upon loading even a single one ...

**Comment by eguiraud** [comment]
The `#ifndef __CLING__` was really needed, you can easily double-check.

About whether we should have `main`s in the tutorials: `git grep 'int main' $ROOTSYS/tutorials` tells me I'm not the first one to think it's useful :smile: If we want a general guideline on this I can bring up the topic at the next team meeting.

**Comment by pcanal** [comment]
main: Yes, we should.  It is indeed currently very inconsistent (none, by-itself, surrounded by arbitrary `#if[n]def`, hidding by `__CLING__`)

File: tutorials/v7/ntuple/ntpl004_dimuon.C [hunk 1]
```diff
@@ -20,7 +20,9 @@
 // Until C++ runtime modules are universally used, we explicitly load the ntuple library.  Otherwise
 // triggering autoloading from the use of templated types would require an exhaustive enumeration
 // of "all" template instances in the LinkDef file.
+#ifdef __CLING__
 R__LOAD_LIBRARY(ROOTNTuple)
+#endif
```
Comment (pcanal) [inline tutorials/v7/ntuple/ntpl004_dimuon.C:25]:
```suggestion
#include <Rtypes.h>

R__LOAD_LIBRARY(ROOTNTuple)
```

**Comment by pcanal** [comment]
> The #ifndef __CLING__ was really needed, you can easily double-check.

Right ... this is because it is not preceded by `#include <Rtypes.h>

**Comment by jblomer** [comment]
> main: Yes, we should. It is indeed currently very inconsistent (none, by-itself, surrounded by arbitrary `#if[n]def`, hidding by `__CLING__`)

Yes we should have guidelines?  I'm in principle fine with the change but I think we should clarify whether or not we want `main` in the tutorials.  If we do, we should change all the RNTuple tutorials.

**Comment by eguiraud** [comment]
We were going to make a decision today at the team meeting but we ran long, so it was put off to next week's team meeting. I'll make this PR a draft until then.

**Pull request closed without merging**